### PR TITLE
Test nb_client module

### PR DESCRIPTION
**Pull request by mpacer**
This add unit tests for the record types and simple functions in nb_client.

It will either take some clever mocking or proper integration tests to properly handle the rest of the classes in this module.

Going through this made me wonder if we should split this module into an nb_types and and nb_client module… as this seemed to be a fairly clean delineation between the two types of functionality.

### CODE REVIEW

**Comment by mpacer** [comment]
There's always something super disappointing when you see a code coverage report for a testing PR that reports a drop in test coverage for some reason…

I don't get it — the tests get picked up by pytest-cov locally:
```
---------- coverage: platform darwin, python 3.6.8-final-0 -----------
Name                               Stmts   Miss  Cover
------------------------------------------------------
bookstore/__init__.py                 10      1    90%
bookstore/_version.py                 10      0   100%
bookstore/archive.py                  49     11    78%
bookstore/bookstore_config.py         17      0   100%
bookstore/client/__init__.py           1      0   100%
bookstore/client/nb_client.py        112     25    78%
bookstore/client/store_client.py      26     18    31%
bookstore/clone.py                    82     18    78%
bookstore/handlers.py                 28     14    50%
bookstore/publish.py                  38     25    34%
bookstore/s3_paths.py                  9      0   100%
bookstore/utils.py                    12      0   100%
bookstore/validation.py               12     12     0%
------------------------------------------------------
TOTAL                                406    124    69%
```

File: bookstore/client/tests/test_nb_client.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    assert kernel_info.id == kernel_info_dict['id']
+    assert kernel_info.name == kernel_info_dict["name"]
+    assert kernel_info.last_activity == kernel_info_dict["last_activity"]
+    assert kernel_info.execution_state == kernel_info_dict['execution_state']
+    assert kernel_info.connections == kernel_info_dict["connections"]
+    assert kernel_info == KernelInfo(**kernel_info_dict)
+
+
+def test_notebook_session_class(notebook_session, session_dict):
+    # assert notebook_session.id == session_dict['id'] # TODO: once id is re-added to NotebookSession's attributes readd this
```
Comment (willingc) [inline bookstore/client/tests/test_nb_client.py]: Let's remove the comments in this file and the other and instead open an issue to track.

File: bookstore/client/tests/test_nb_client.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            "f92b7c8b-0858-4d10-903c-b0631540fb36",
+        ),
+        (
+            "kernel-ee2b7c8b-0858-4d10-903c-b0631540fb36.json",
+            "ee2b7c8b-0858-4d10-903c-b0631540fb36",
+        ),
+    ],
+)
+def test_extract_kernel_id(connection_file, expected_kernel_id):
+    assert expected_kernel_id == extract_kernel_id(connection_file)
```
Comment (willingc) [inline bookstore/client/tests/test_nb_client.py]: Perhaps flip the assert statement. `assert <test_result> <expected result>`

File: tox.ini [hunk 3]
```diff
@@ -73,7 +73,7 @@ deps =
 depends =
     py36: clean
 commands =
-    pytest -v --maxfail=2 --cov-config=.coveragerc --cov=bookstore -W always bookstore/tests/
+    pytest -v --maxfail=2 --cov-config=.coveragerc --cov=bookstore -W always
```
Comment (willingc) [inline tox.ini]: I would be inclined to leave the directory here and move the client tests to `bookstore/tests/client/...`

Comment (willingc) [inline tox.ini]: I would be inclined to leave the directory here and move the client tests to `bookstore/tests/client/...`

Comment (willingc) [inline tox.ini]: I would be inclined to leave the directory here and move the client tests to `bookstore/tests/client/...`

File: bookstore/client/nb_client.py [hunk 4]
```diff
@@ -78,16 +78,29 @@ def __init__(self, *args, id, name, last_activity, execution_state, connections)
     def __repr__(self):
         return json.dumps(self.model, indent=2)
 
+    def __eq__(self, other):
+        if isinstance(other, KernelInfo):
```
[the remark this answers is not in the document]
Comment (mpacer) [reply bookstore/client/nb_client.py]: Thanks!

File: bookstore/client/tests/test_nb_client.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
+import pytest
```
[the remark this answers is not in the document]
Comment (mpacer) [reply bookstore/client/tests/test_nb_client.py]: Sounds good

File: bookstore/client/tests/test_nb_client.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+from bookstore.client.nb_client import (
+    NotebookSession,
+    KernelInfo,
+    extract_kernel_id,
+    LiveNotebookRecord,
+)
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (mpacer) [reply bookstore/client/tests/test_nb_client.py]: Sounds good

File: bookstore/client/tests/test_nb_client.py (hunk 2 above)
Comment (mpacer) [reply bookstore/client/tests/test_nb_client.py]: Sounds good

File: tox.ini (hunk 3 above)
Comment (mpacer) [reply tox.ini]: Ok.

File: bookstore/client/tests/test_nb_client.py (hunk 1 above)
Comment (mpacer) [reply bookstore/client/tests/test_nb_client.py]: I already created the issue.

**Comment by mpacer** [comment]
Thank you for the reviews @willingc! I think I addressed them all, please let me know if not.

**Comment by willingc** [comment]
Fantastic @mpacer. Happy to merge.

**Pull request merged**